File: sdks/php/src/Event.php
<?php

declare(strict_types=1);

namespace Aeon\ProcessorSdk;

/** Inbound event from the Aeon engine. */
final class Event
{
    /** @param list<array{0: string, 1: string}> $metadata */
    public function __construct(
        public readonly string $id,
        public readonly int $timestamp,
        public readonly string $source,
        public readonly int $partition,
        public readonly array $metadata,
        public readonly string $payload,
        public readonly ?int $sourceOffset = null,
    ) {}

    public function payloadString(): string
    {
        return $this->payload;
    }
}

File: sdks/java/src/main/java/io/aeon/processor/Event.java
package io.aeon.processor;

import java.nio.charset.StandardCharsets;
import java.util.List;

/**
 * Canonical Aeon event envelope for the Java Processor SDK.
 */
public record Event(
    String id,
    long timestamp,
    String source,
    int partition,
    List<String[]> metadata,
    byte[] payload,
    Long sourceOffset
) {
    /**
     * Return payload decoded as UTF-8 string.
     */
    public String payloadString() {
        return new String(payload, StandardCharsets.UTF_8);
    }
}
